### ISSUE TITLE
Reopening extension issue - problem with Firefox

### ISSUE THREAD

**Issue by dornech** [GCuser99/SeleniumVBA#118]
Hi there, sorry, borthering you again with the extension issue. 

AddExtension throws an error when using FireFox
![image](https://github.com/user-attachments/assets/b38a1859-7ad3-4872-b3ed-ab37f6c9b904)
This irregardless if passing on a base64 encoded string/bytearray or a path (tested with SeleniumVBA 5.4 and 5.5). Testing different versions works quite easily using the switch described here https://github.com/GCuser99/SeleniumVBA/discussions/115.

As I work with Selenium in Python as well, i think i found the reason: Firefox  / geckodriver does not allow installation of extensions via Capabilities/Options anymore but you have to use instal-addon in the driver
https://www.selenium.dev/selenium/docs/api/py/_modules/selenium/webdriver/firefox/firefox_profile.html#FirefoxProfile.add_extensionhttps://www.selenium.dev/selenium/docs/api/py/_modules/selenium/webdriver/firefox/webdriver.html#WebDriver.install_addon

**Comment by dornech** [comment]
I tested - it works with 
```
objJSONparam.Add "addon", encodeBase64(readFileToBytearray(s_BrowserExtensionFile))
objWebDriver.ExecuteCmd "POST", "/session/$sessionId/moz/addon/install", objJSONconverter.ConvertToJson(objJSONparam)
```
where objJSONparam is a dictionary.
So, I think for AddExtension you should at least install a browser check. 
Optionally you can add a install-method  for FireFox but also including a browser check.

**Comment by GCuser99** [comment]
Thanks for doing the research on this - I'll include in next update.

Before we put this to rest - you said previously that you had what you thought was a memory issue with the encoding of the Ad Blocker extension in Chrome. Did you verify that the problem was happening with encoding, and not with the WebDriver? If it is happening with encoding, then have you tried doing the encoding in bytes chunks to avoid the issue? I had no problem encoding on my 64-bit system.

**Comment by dornech** [comment]
I think this issue was probably caused by IDE debugging limitations. 
To be honest: I work for this stuff on a machine with Office 2003 (BTW: SeleniumVBA works with that as well - only issue for the VBA version are the 64bit compatible Declares - but that can be fixed easily (LongPtr -> Long, LongLong -> Long).

**Comment by GCuser99** [comment]
Fixed in v5.6 - added new method `InstallAddons` of WebDriver class (FF only).

**Comment by GCuser99** [comment]
> I think this issue was probably caused by IDE debugging limitations. To be honest: I work for this stuff on a machine with Office 2003 (BTW: SeleniumVBA works with that as well - only issue for the VBA version are the 64bit compatible Declares - but that can be fixed easily (LongPtr -> Long, LongLong -> Long).

BTW I forgot to ask you... In twinBASIC, VBA7-type Windows API declares (with PtrSafe) do not need to be altered to compile in 32-bit. In other words, the same (VBA7-like) code base can be compiled in either 32- or 64-bit. So "in theory", 32-bit DLL could possibly work in pre-2007 versions of Office. Can you confirm that the ActiveX DLL works without modification on your Office 2003? I have no way of testing that.

**Comment by dornech** [comment]
Yes, I can confirm that the DLL works without modification for Office2003. Thaught this was clear form the context. I hat some difficulties getting used to the ribbon stuff so I was not too eager to switch Office version - and it worked so well. In general I am quite positvie about the compatibility between different Office versions.

The hint regarding changing Declares was with regard to using the VBA macro version as Office add-In. I took the xlsm file, changed the declares and stored as xla / Excel add-in and it works perfectly when switchting the project reference. Compared to SeleniumBasic I like to be able to switch in order to have more debugging insights with the xla based solution. Access I did not try yet but would not expect it not to work the same way.
